i18n

digitall-it opened the pull request:
Didn't test the code, fixes #16.

digitall-it committed:
Create en.yml

digitall-it committed:
Create it.yml

digitall-it committed:
Update en.yml

digitall-it committed:
Update it.yml

digitall-it committed:
Update ShareCare.php

digitall-it committed:
Update ShareCareFields.php

digitall-it committed:
Update en.yml

digitall-it committed:
Update en.yml

digitall-it committed:
Update en.yml

digitall-it committed:
Update ShareCareSingleSummary.php

digitall-it committed:
Update ShareCarePreview.ss

digitall-it committed:
Update ShareCarePreview.ss

digitall-it committed:
Update it.yml

lerni commented:
I've also created a translation (de), but wasn't ready to PR. Unfortunately @digitall-it PR crashes because of static $cms_message is set to the translation-string (https://github.com/digitall-it/silverstripe-share-care/blob/master/code/ShareCare.php#L43). I think "$cms_message" should be removed and "_t"-translation should be used - right?

Do you @jonom prefer uppercase translation keys?

I also think it would make sense to add default lang-strings to the _t-functions, so you can auto-generate en.yml

I would be willing to merge @digitall-it italian and my german translation into one PR.

digitall-it commented:
That was my only fear, that statics and translation strings wouldn't mix.

> I think `"$cms_message"` should be removed and `"_t"`-translation should be used

How to do that?

lerni commented:
@digitall-it I would just reference _t('ShareCare.CMSMessage') instead of `$cms_message` where ever it is used.

jonom commented on code/ShareCare.php:
```diff
@@ -40,7 +40,7 @@ class ShareCare extends DataExtension
      * @config
      */
-    private static $cms_message = 'When this page is shared by people on social media it will look something like this:';
+    private static $cms_message = _t('ShareCare.CMSMessage');
```
As @lerni suggested, this line can be deleted, and wherever `$cms_message` was used, use `_t('ShareCare.CMSMessage')` instead.

jonom commented on code/ShareCare.php:
```diff
@@ -49,10 +49,10 @@ public function updateCMSFields(FieldList $fields)
         $msg = Config::inst()->get('ShareCare', 'cms_message');
         if ($msg) {
-            $fields->addFieldToTab('Root.Share', new LiteralField('ShareCareMessage',
+            $fields->addFieldToTab(_t('ShareCare.TabName'), new LiteralField('ShareCareMessage',
```
Please add a default EN string to each `_t()` call, e.g. `_t('ShareCare.TabName', 'Share')` here.

jonom commented on code/ShareCare.php:
```diff
@@ -197,8 +197,8 @@ public function EmailShareLink()
-        $subject = rawurlencode('Thought you might like this');
-        $body = rawurlencode("Thought of you when I found this: $pageURL");
+        $subject = rawurlencode(_t('ShareCare.EmailSubject'));
+        $body = rawurlencode(_t('ShareCare.Body') . " " . $pageURL);
```
Can you please revise this to pass the $pageURL variable in to the translation function. See third example here: https://docs.silverstripe.org/en/3/developer_guides/i18n/#usage-in-php-files

jonom commented on code/ShareCareFields.php:
```diff
@@ -30,24 +30,24 @@ public function updateCMSFields(FieldList $fields)
             ->setAllowedFileCategories('image')
             ->setAllowedMaxFileNumber(1)
-            ->setDescription('<a href="https://developers.facebook.com/docs/sharing/best-practices#images" target="_blank">Optimum image ratio</a> is 1.91:1. (1200px wide by 630px tall or better)'));
+            ->setDescription(_t('ShareCareFields.ShareImageRation')));
```
Typo here - can you please remove the trailing `n` so it is `ShareImageRatio`

jonom commented on code/ShareCareFields.php:
```diff
@@ -30,24 +30,24 @@ public function updateCMSFields(FieldList $fields)
+            ->setDescription(_t('ShareCareFields.ShareImageRation')));
         if (Config::inst()->get('ShareCare', 'pinterest')) {
-            $fields->addFieldToTab('Root.Share', UploadField::create('PinterestImageCustom', 'Pinterest image')
+            $fields->addFieldToTab(_t('ShareCare.TabName'), UploadField::create('PinterestImageCustom', _t('ShareCareFields.SharePinterestImage'))
```
Can we shorten this to `ShareCareFields.PinterestImage` please

jonom commented on code/ShareCareFields.php:
```diff
@@ -30,24 +30,24 @@ public function updateCMSFields(FieldList $fields)
                 ->setAllowedFileCategories('image')
                 ->setAllowedMaxFileNumber(1)
-                ->setDescription('Square/portrait or taller images look best on Pinterest. This image should be at least 750px wide.'));
+                ->setDescription(_t('ShareCareFields.SharePinterestDescription')));
```
Shorten to `ShareCareFields.PinterestImageDescription` please

jonom commented on code/ShareCareSingleSummary.php:
```diff
@@ -23,15 +23,15 @@ public function updateCMSFields(FieldList $fields)
         // Add summary fields
-        $fields->addFieldToTab('Root.Main', TextAreaField::create('MetaDescription', 'Content summary')
-            ->setDescription('Summarise the content of this page. This will be used for search engine results and social media so make it enticing.')
+        $fields->addFieldToTab('Root.Main', TextAreaField::create('MetaDescription', _t('ShareCareSummary.DescriptionTitle'))
```
I think I like `ShareCareSummary.SummaryTitle` better

jonom commented on code/ShareCareSingleSummary.php:
```diff
@@ -23,15 +23,15 @@ public function updateCMSFields(FieldList $fields)
-        $fields->addFieldToTab('Root.Main', TextAreaField::create('MetaDescription', 'Content summary')
-            ->setDescription('Summarise the content of this page. This will be used for search engine results and social media so make it enticing.')
+        $fields->addFieldToTab('Root.Main', TextAreaField::create('MetaDescription', _t('ShareCareSummary.DescriptionTitle'))
+            ->setDescription(_t('ShareCareSummary.DescriptionDescription'))
```
So this one can be `ShareCareSummary.SummaryDescription`

jonom commented on templates/ShareCarePreview.ss:
```diff
@@ -84,14 +84,14 @@
 
 	</div>
-	<span class="description">Previews will be updated after saving. Changes need to be published before they take effect.</span>
+	<span class="description"><%t ShareCare.Description "Previews will be updated after saving. Changes need to be published before they take effect." %></span>
```
Can we call this `ShareCare.SaveNotice` please

jonom commented on code/ShareCareFields.php:
```diff
@@ -21,7 +21,7 @@ class ShareCareFields extends DataExtension
      * @config
      */
-    private static $cms_message = 'The preview above is automatically generated from your content. You can override the default values using these fields:';
+    private static $cms_message = _t('ShareCareFields.CMSMessage');
```
As with previous, please remove this static and reference the translatable string instead in code.

jonom reviewed (changes requested):
@digitall-it sorry for the late feedback, I somehow missed this PR when it was opened. Generally this looks good thanks, just a few things to tweak if you don't mind.

Commit:
fix Fatal error: Constant expression contains invalid operations in /framework/core/manifest/ConfigStaticManifest.php(399) : eval()'d code on line 1 - don't assign _t to static,
remove static $cms_message add it to lang-files,
Link in ShareCareFields.ShareImageRatio in ShareCareFields.php and not in all lang-files,
double quotes in lang files prevent escaping issues,
add de.yml,
reword ShareCareFields.CMSMessage - do not use "above", since extension-order can differ

digitall-it committed:
Merge pull request #1 from lerni/master

fix crash, add de.yml

Commit:
should cover all requests jonom made

digitall-it committed:
Merge pull request #2 from lerni/master

should now cover all requests jonom made

jonom commented on code/ShareCare.php:
```diff
@@ -35,24 +35,17 @@ class ShareCare extends DataExtension
     public function updateCMSFields(FieldList $fields)
     {
-        $msg = Config::inst()->get('ShareCare', 'cms_message');
+		$msg = _t('ShareCare.CMSMessage');
```
1. Please move the default string 'When this page is shared by people on social media it will look something like this:' from $tab to $msg. Default for $tab is 'Share'.
2. Spaces for indentation please on PHP files. A few tabs have been introduced in ShareCareFields.php and ShareCare.php (just find/replace with 4 spaces).

jonom commented on code/ShareCare.php:
```diff
@@ -197,8 +190,8 @@ public function EmailShareLink()
-        $subject = rawurlencode('Thought you might like this');
-        $body = rawurlencode("Thought of you when I found this: $pageURL");
+        $subject = rawurlencode(_t('ShareCare.EmailSubject','Thought you might like this'));
+        $body = rawurlencode(_t('ShareCare.EmailBody','Thought of you when I found this: {URL}', [URL=>$pageURL]));
```
This causes an error because URL should be wrapped in quotes. Can we go a step further and use PHP5.3 syntax as that's the minimum requirement for SS3.x. `array('URL' => $pageURL)`

jonom commented on code/ShareCareFields.php:
```diff
@@ -16,38 +16,31 @@ class ShareCareFields extends DataExtension
             ->setAllowedFileCategories('image')
             ->setAllowedMaxFileNumber(1)
-            ->setDescription('<a href="https://developers.facebook.com/docs/sharing/best-practices#images" target="_blank">Optimum image ratio</a> is 1.91:1. (1200px wide by 630px tall or better)'));
+			->setDescription(_t('ShareCareFields.ShareImageRatio','{Link}Optimum image ratio</a> is 1.91:1. (1200px wide by 630px tall or better)', ['Link'=>'<a href="https://developers.facebook.com/docs/sharing/best-practices#images" target="_blank">'])));
```
PHP 5.3 array syntax please

jonom reviewed (changes requested):
Thanks guys. Couple more little things to fix then I'll merge in.

digitall-it committed:
Fix switched i18n defaults on CMS fields on ShareCare, fix wrong identation, fix php 5.3 style arrays

digitall-it commented:
Thanks to both for helping me learn how to care to share share-care.

jonom commented:
Thanks @digitall-it and @lerni for this contribution!

jonom commented:
Shipped in [2.0](https://github.com/jonom/silverstripe-share-care/releases/tag/v2.0.0)
